docs/collisions.md

```diff
@@ -64,6 +64,62 @@ For example if a folder has five 5 with the same name say **_A.txt _** the five
 * [Extensions in folder names matter](https://apple.stackexchange.com/questions/123001/renamed-folder-becomes-a-file-with-an-extension) and should be handled :
 * Case sensitivity is taken into account. `Myfile` and `myFile` are different for example. That is useful to [ensure complete syncs](https://rclone.org/overview/#case-insensitive).
 
+#### Extensions
+
+For both files and folders the extension is any string that follows after the last dot.
+
+For example:
+<table>
+   <thead>
+      <th>File/Folder Name</th>
+      <th>Extension</th>
+      <th>Remark</th>
+   </thead>
+   <tbody>
+      <tr>
+         <td>My File.txt</td>
+         <td>txt</td>
+         <td>Regular</td>
+      </tr>
+      <tr>
+         <td>My File.txt(1)</td>
+         <td>txt(1)</td>
+         <td>Extension contains non-alphanumeric chars</td>
+      </tr>
+      <tr>
+         <td>My File.app</td>
+         <td>app</td>
+         <td>Regular</td>
+      </tr>
+      <tr>
+         <td>something.something</td>
+         <td>something</td>
+         <td>Name same as extension</td>
+      </tr>
+      <tr>
+         <td>2022.03.02</td>
+         <td>02</td>
+         <td>Extension is number</td>
+      </tr>
+      <tr>
+         <td>Folder.Work Files</td>
+         <td>Work Files</td>
+         <td>Extension contains spaces</td>
+      </tr>
+   </tbody>
+</table>
+
+As seen, extension can take infinite number of "shapes" BUT as long as all characters in the file name including the extension are valid file name chacters, then, **the extension is the string that follows after the last dot** (Emphasis).
```

Comment:
This raises the question of files that contain characters which are *not* valid in a filename -- but doesn't answer that question :-).  Can we ever have such characters, or do we disallow them in the first place?  If they're not allowed and can never be there, then we don't need to address them.  If we might encounter them, then that's a separate problem, and maybe not one to be documented here?  It's not really related to collision handling.

Also, in general I don't think we need all these examples.  If you just say "The extension is everything following the last dot (".", ASCII/Unicode 46) in the filename, and if the filename has no dots then it has no extension." that's clear, totally unambiguous, and it's all we need to say.  You could add something like this: "This means that an extension might contain spaces, parentheses, and other characters not traditionally thought of as extension characters.  It also means that the last dot could also be the first dot, in which case the extension is the remaining string after that first dot." if you want to make sure people understand that we really, literally mean everything after the last dot.

Reply:
Thanks @kfogel this is really good! Done.